pkg/email/email_test.go

```diff
@@ -0,0 +1,77 @@
+package email
+
+import (
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/appconfig"
+	"github.com/cmsgov/easi-app/pkg/testhelpers"
+)
+
+type SESTestSuite struct {
+	suite.Suite
+	logger *zap.Logger
+	config Config
+}
+
+type mockSender struct {
+	toAddress string
+	subject   string
+	body      string
+}
+
+func (s *mockSender) Send(toAddress string, subject string, body string) error {
+	s.toAddress = toAddress
+	s.subject = subject
+	s.body = body
+	return nil
+}
+
+type mockFailedSender struct{}
+
+func (s *mockFailedSender) Send(toAddress string, subject string, body string) error {
+	return errors.New("sender had an error")
+}
+
+type mockFailedTemplateCaller struct{}
+
+func (c mockFailedTemplateCaller) Execute(wr io.Writer, data interface{}) error {
+	return errors.New("template caller had an error")
+}
+
+func TestSESTestSuite(t *testing.T) {
+	logger := zap.NewNop()
+	config := testhelpers.NewConfig()
+
+	emailConfig := Config{
+		GRTEmail:          config.GetString(appconfig.GRTEmailKey),
+		URLHost:           config.GetString(appconfig.ApplicationHostKey),
+		URLScheme:         config.GetString(appconfig.ApplicationProtocolKey),
+		TemplateDirectory: config.GetString(appconfig.EmailTemplateDirectoryKey),
+	}
+
+	// we'll need this code for integration testing
```

Comment:
I tested this locally, but it really belong in eventually integration tests/server instantiation. This will be a follow-up PR.